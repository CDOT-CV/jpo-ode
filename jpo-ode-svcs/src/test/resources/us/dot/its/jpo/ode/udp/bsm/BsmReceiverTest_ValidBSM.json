{
  "metadata": {
    "bsmSource": "EV",
    "logFileName": "bsmTx.gz",
    "recordType": "bsmTx",
    "securityResultCode": "success",
    "receivedMessageDetails": {
      "locationData": {
        "latitude": "40.5657881",
        "longitude": "-105.0316742",
        "elevation": "1489",
        "speed": "0.4",
        "heading": "267.4"
      },
      "rxSource": "NA"
    },
    "payloadType": "us.dot.its.jpo.ode.model.OdeBsmPayload",
    "serialId": {
      "streamId": "801780cb-d91d-444b-8f4d-9c44fe27f5ea",
      "bundleSize": 16,
      "bundleId": 72,
      "recordId": 15,
      "serialNumber": 14741
    },
    "odeReceivedAt": "2019-04-09T18:07:47.798Z",
    "schemaVersion": 6,
    "recordGeneratedAt": "2018-05-01T15:55:55.494Z",
    "recordGeneratedBy": "OBU",
    "sanitized": false
  },
  "payload": {
    "dataType": "us.dot.its.jpo.ode.plugin.j2735.J2735Bsm",
    "data": {
      "coreData": {
        "msgCnt": 119,
        "id": "31325433",
        "secMark": 55494,
        "position": {
          "latitude": 40.5657881,
          "longitude": -105.0316742,
          "elevation": 1489.0
        },
        "accelSet": {
          "accelLat": 0.00,
          "accelLong": 0.43,
          "accelVert": 0.00,
          "accelYaw": 0.00
        },
        "accuracy": {
          "semiMajor": 9.30,
          "semiMinor": 12.05
        },
        "transmission": "NEUTRAL",
        "speed": 0.40,
        "heading": 267.4000,
        "brakes": {
          "wheelBrakes": {
            "leftFront": false,
            "rightFront": false,
            "unavailable": true,
            "leftRear": false,
            "rightRear": false
          },
          "traction": "unavailable",
          "abs": "unavailable",
          "scs": "unavailable",
          "brakeBoost": "unavailable",
          "auxBrakes": "unavailable"
        },
        "size": {
          "width": 190,
          "length": 570
        }
      },
      "partII": [
        {
          "id": "VehicleSafetyExtensions",
          "value": {
            "pathHistory": {
              "crumbData": [
                {
                  "elevationOffset": 1.0,
                  "latOffset": 0.0000051,
                  "lonOffset": 0.0000097,
                  "timeOffset": 0.90
                },
                {
                  "elevationOffset": 4.7,
                  "latOffset": 0.0000141,
                  "lonOffset": 3E-7,
                  "timeOffset": 3.69
                },
                {
                  "elevationOffset": 9.4,
                  "latOffset": 0.0000341,
                  "lonOffset": 0.0000326,
                  "timeOffset": 8.29
                },
                {
                  "elevationOffset": 13.5,
                  "latOffset": 0.0000438,
                  "lonOffset": 0.0000839,
                  "timeOffset": 18.29
                },
                {
                  "elevationOffset": 12.0,
                  "latOffset": 0.0000484,
                  "lonOffset": 0.0001072,
                  "timeOffset": 26.49
                },
                {
                  "elevationOffset": 11.7,
                  "latOffset": 0.0000478,
                  "lonOffset": 0.0001117,
                  "timeOffset": 28.00
                },
                {
                  "elevationOffset": 11.3,
                  "latOffset": 0.0000320,
                  "lonOffset": 0.0001211,
                  "timeOffset": 28.80
                },
                {
                  "elevationOffset": 11.2,
                  "latOffset": 0.0000301,
                  "lonOffset": 0.0001197,
                  "timeOffset": 30.00
                },
                {
                  "elevationOffset": 7.8,
                  "latOffset": 0.0000202,
                  "lonOffset": 0.0001092,
                  "timeOffset": 34.90
                },
                {
                  "elevationOffset": 1.9,
                  "latOffset": 0.0000230,
                  "lonOffset": 0.0001186,
                  "timeOffset": 40.00
                },
                {
                  "elevationOffset": -0.1,
                  "latOffset": 0.0000248,
                  "lonOffset": 0.0001401,
                  "timeOffset": 42.89
                },
                {
                  "elevationOffset": -3.3,
                  "latOffset": 0.0000343,
                  "lonOffset": 0.0001451,
                  "timeOffset": 49.80
                },
                {
                  "elevationOffset": -3.6,
                  "latOffset": 0.0000443,
                  "lonOffset": 0.0001437,
                  "timeOffset": 51.60
                },
                {
                  "elevationOffset": -6.9,
                  "latOffset": 0.0000466,
                  "lonOffset": 0.0001419,
                  "timeOffset": 56.49
                },
                {
                  "elevationOffset": -8.4,
                  "latOffset": 0.0000346,
                  "lonOffset": 0.0001261,
                  "timeOffset": 62.49
                }
              ]
            },
            "pathPrediction": {
              "confidence": 0.0,
              "radiusOfCurve": 0.0
            }
          }
        },
        {
          "id": "SupplementalVehicleExtensions",
          "value": {}
        }
      ]
    }
  }
}