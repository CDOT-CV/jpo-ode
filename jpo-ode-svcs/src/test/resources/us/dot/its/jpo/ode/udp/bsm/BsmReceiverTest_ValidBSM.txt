001480845d84abbc80cfeb26aa81a01638ff9ab3c70f8abed322fb36b87e7e770eb07ea005400000020104c0c0f1b0c028705c02180e9ebc072109403460e3e2410150b604360de3ac1e6b0dc05260d8b9433df10e062a0d40344eb3144072e0ce69c7baf18008a8708d8c002127c0009000000fdfffe3ffff9407346004000041910120100000000000001dabc3c90302092101c73bd0503ffffc102843d31d0b1fffc9f8a8c01064404804000000849f00024000003f7fff8ffffe501ceccc100000012010