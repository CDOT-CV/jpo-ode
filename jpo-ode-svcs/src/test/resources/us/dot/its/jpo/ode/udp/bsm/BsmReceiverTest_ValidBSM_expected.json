{
  "metadata": {
    "bsmSource": "EV",
    "recordType": "bsmTx",
    "securityResultCode": "success",
    "receivedMessageDetails": {
      "locationData": {
        "latitude": "unavailable",
        "longitude": "unavailable",
        "elevation": "unavailable",
        "speed": "unavailable",
        "heading": "unavailable"
      },
      "rxSource": "RSU"
    },
    "payloadType": "us.dot.its.jpo.ode.model.OdeAsn1Payload",
    "serialId": {
      "streamId": "de02c4a6-9ae3-4417-8bbc-636119ac0767",
      "bundleSize": 1,
      "bundleId": 0,
      "recordId": 0,
      "serialNumber": 0
    },
    "odeReceivedAt": "2024-11-26T23:53:21.120Z",
    "schemaVersion": 7,
    "maxDurationTime": 0,
    "recordGeneratedBy": "OBU",
    "sanitized": false,
    "asn1": "303031347B0A2020226D65746164617461223A207B0A202020202262736D536F75726365223A20224556222C0A20202020226C6F6746696C654E616D65223A202262736D54782E677A222C0A20202020227265636F726454797065223A202262736D5478222C0A20202020227365637572697479526573756C74436F6465223A202273756363657373222C0A202020202272656365697665644D65737361676544657461696C73223A207B0A202020202020226C6F636174696F6E44617461223A207B0A2020202020202020226C61746974756465223A202234302E35363537383831222C0A2020202020202020226C6F6E676974756465223A20222D3130352E30333136373432222C0A202020202020202022656C65766174696F6E223A202231343839222C0A2020202020202020227370656564223A2022302E34222C0A20202020202020202268656164696E67223A20223236372E34220A2020202020207D2C0A202020202020227278536F75726365223A20224E41220A202020207D2C0A20202020227061796C6F616454797065223A202275732E646F742E6974732E6A706F2E6F64652E6D6F64656C2E4F646542736D5061796C6F6164222C0A202020202273657269616C4964223A207B0A2020202020202273747265616D4964223A202238303137383063622D643931642D343400",
    "originIp": "127.0.0.1"
  },
  "payload": {
    "dataType": "us.dot.its.jpo.ode.model.OdeHexByteArray",
    "data": {
      "bytes": "303031347B0A2020226D65746164617461223A207B0A202020202262736D536F75726365223A20224556222C0A20202020226C6F6746696C654E616D65223A202262736D54782E677A222C0A20202020227265636F726454797065223A202262736D5478222C0A20202020227365637572697479526573756C74436F6465223A202273756363657373222C0A202020202272656365697665644D65737361676544657461696C73223A207B0A202020202020226C6F636174696F6E44617461223A207B0A2020202020202020226C61746974756465223A202234302E35363537383831222C0A2020202020202020226C6F6E676974756465223A20222D3130352E30333136373432222C0A202020202020202022656C65766174696F6E223A202231343839222C0A2020202020202020227370656564223A2022302E34222C0A20202020202020202268656164696E67223A20223236372E34220A2020202020207D2C0A202020202020227278536F75726365223A20224E41220A202020207D2C0A20202020227061796C6F616454797065223A202275732E646F742E6974732E6A706F2E6F64652E6D6F64656C2E4F646542736D5061796C6F6164222C0A202020202273657269616C4964223A207B0A2020202020202273747265616D4964223A202238303137383063622D643931642D343400"
    }
  }
}